src/api/entities/confidential/ConfidentialAccount/index.ts

```diff
@@ -87,4 +90,24 @@ export class ConfidentialAccount extends Entity<UniqueIdentifiers, string> {
   public toHuman(): string {
     return this.publicKey;
   }
+
+  /**
+   * Retrieve the ConfidentialAssets associated to this Account
+   *
+   * @note uses the middlewareV2
+   * @note there is a possibility that the data is not ready by the time it is requested. In that case, `null` is returned
+   */
+  public async getHeldAssets(): Promise<string[]> {
```

Comment:
```suggestion
  public async getHeldAssets(
    opts: {
      order?: AssetHoldersOrderBy;
      size?: BigNumber;
      start?: BigNumber;
    } = {}
  ): Promise<ResultSet<ConfidentialAsset>> {
```

looking at the Identity signature, I think we should allow pagination options + return the ConfidentialAsset entity